tests/nosetests/pyanaconda_tests/module_source_cdrom_test.py

```diff
@@ -0,0 +1,310 @@
+
+        device_tree = Mock()
+
+        device_tree.FindOpticalMedia = Mock()
+        device_tree.FindOpticalMedia.return_value = [device.name for device in devices]
+
+        device_tree.GetDeviceData = Mock()
+        device_tree.GetDeviceData.side_effect = [DeviceData.to_structure(dev) for dev in devices]
+
+        return device_tree
+
+    def assert_resolve_and_mount_calls(self,
+                                       device_tree_mock, mount_mock,
+                                       num_called, num_untouched):
+        """Check that a given number of mock CD-ROMs was accessed
+
+        All devices in the supplied range must have been resolved and mounted (tried to, anyway)
+        with GetDeviceData() and mount() respectively. The rest must have been not. The assumption
+        is that the called ones precede the untouched ones, because finding a match skips the rest.
+        This matches the logic in tested method.
+        """
+        for n in range(num_called):
+            self.assertIn(
+                call("test{}".format(n)),
+                device_tree_mock.GetDeviceData.mock_calls
+            )
+            self.assertIn(
+                call("/dev/cdrom-test{}".format(n), "/mnt/put-cdrom-here", "iso9660", "ro"),
+                mount_mock.mock_calls
+            )
+
+        for n in range(num_called, num_called + num_untouched):
+            self.assertNotIn(
+                call("test{}".format(n)),
+                device_tree_mock.GetDeviceData.mock_calls
+            )
+            self.assertNotIn(
+                call("/dev/cdrom-test{}".format(n), "/mnt/put-cdrom-here", "iso9660", "ro"),
+                mount_mock.mock_calls
+            )
+
+        self.assertEqual(device_tree_mock.GetDeviceData.call_count, num_called)
+        self.assertEqual(mount_mock.call_count, num_called)
+
+    @patch("pyanaconda.modules.payloads.source.cdrom.initialization._is_valid_install_disk")
+    @patch("pyanaconda.modules.payloads.source.cdrom.initialization.unmount")
+    @patch("pyanaconda.modules.payloads.source.cdrom.initialization.mount")
+    @patch_dbus_get_proxy
+    def run_choice_test(self, proxy_getter, mount_mock, unmount_mock, valid_mock):
+        """Test CD-ROM Source setup installation source task run - everything."""
+        # Fake four CD-ROM devices: First fails to mount, second has nothing useful, third has what
+        # we want so is left mounted, fourth is entirely skipped.
+        # The other two tests are needed only to test the exit when nothing is found.
+
+        device_tree = self.set_up_device_tree(4)
+        proxy_getter.return_value = device_tree
+
+        mount_mock.side_effect = \
+            [MountFilesystemError("Mocked failure"), DEFAULT, DEFAULT, DEFAULT]
+
+        # only for devices 2-4; the expected first call is prevented by the exception from mount
+        valid_mock.side_effect = [False, True, False]
+
+        task = SetUpCdromSourceTask("/mnt/put-cdrom-here")
```

Comment:
This string is used between different method. It should be passed in from one place or a constant.

Reply:
Done.